### DISCUSSION TITLE
ManyRelatedField does not seem to run validators of child relation

### DISCUSSION THREAD

**Discussion by dtomas** [encode/django-rest-framework#8698] [Potential Issue]
When using a `PrimaryKeyRelatedField` with `many=True`, its `validators` are not run on list items.
Example:
```python
def validate_object(obj):
    if not obj.is_valid:
        raise ValidationError("Object is not valid")

class SerializerWithManyPKField(Serializer):
    objects = PrimaryKeyRelatedField(many=True, validators=[validate_object], queryset=SomeModel.objects.all())

obj = SomeModel.objects.create(is_valid=False)
serializer = SerializerWithManyPKField(data={"objects": [obj.id]})
serializer.is_valid() # Returns `True`, does not call `validate_object`
```

(Using a `ListField` with `child=PrimaryKeyRelatedField(validators=[validate_object], queryset=SomeModel.objects.all())` works, though.)

**Comment by lovelydinosaur** [comment]
Interesting.

I can see that the `ListField` implementation doesn't really match up with the `ManyRelatedField` implementation.
I don't know if there's a good reason for that or not.

Here's `ListField`...

https://github.com/encode/django-rest-framework/blob/1fd268ae2b03e3328ba7923a47c6a912b2bc0166/rest_framework/fields.py#L1665-L1676

Which calls into `run_child_validation`...

https://github.com/encode/django-rest-framework/blob/1fd268ae2b03e3328ba7923a47c6a912b2bc0166/rest_framework/fields.py#L1684-L1698

There's no equivalent in `ManyRelatedField`...

https://github.com/encode/django-rest-framework/blob/1fd268ae2b03e3328ba7923a47c6a912b2bc0166/rest_framework/relations.py#L520-L529

I don't know if we're able to change this or if there's reason that'd break existing behaviour. Best I can suggest would be someone working on a pull request to bring `ManyRelatedField` in line with the `ListField` behaviour, and seeing if the test suite still passes.
